pkg/types/query/query.go

```diff
@@ -0,0 +1,294 @@
+package query
+
+import "fmt"
+
+// Visitor should have a per chain per db type implementation that converts primitives to db queries.
+type Visitor interface {
+	ComparerPrimitive(primitive ComparerPrimitive)
```

Comment:
nit/ if these are always going to be Primitives, you can drop the suffix and make a primitives pkg in this `query` dir

Reply:
sgtm